--- a/MediaInfoLib/Source/MediaInfo/Multiple/File_Flv.cpp
+++ b/MediaInfoLib/Source/MediaInfo/Multiple/File_Flv.cpp
@@ -485,7 +485,8 @@
     #endif //MEDIAINFO_DEMUX
 
     //Internal
-    Stream.resize(3); //Null, Video, Audio
+    //Stream.resize(3); //Null, Video, Audio
+    Stream.resize(Stream_Max); // Support all potential stream kinds
 
     //Temp
     Searching_Duration=false;
@@ -621,13 +622,20 @@
         Fill(Stream_General, 0, General_Duration, meta_duration, 0, true);
 
     //Purge what is not needed anymore
-    if (!File_Name.empty()) //Only if this is not a buffer, with buffer we can have more data
-        Stream.clear();
+    //if (!File_Name.empty()) //Only if this is not a buffer, with buffer we can have more data
+    //    Stream.clear();
+    if (!File_Name.empty()) { //Only if this is not a buffer, with buffer we can have more data
+				Stream.clear();
+				Stream.resize(Stream_Max);
+	  }
 }
 
 //---------------------------------------------------------------------------
 void File_Flv::Streams_Finish_PerStream(stream_t StreamKind)
 {
+		if (Stream.size() <= (size_t)StreamKind)
+			return;
+	
     if (Stream[StreamKind].TimeStamp!=(int32u)-1)
     {
         //Calculating the last timestamp (last block included)
